head_bench_test.go

```diff
@@ -0,0 +1,61 @@
+package tsdb
+
+import (
+	"strconv"
+	"sync/atomic"
+	"testing"
+
+	"github.com/prometheus/tsdb/labels"
+	"github.com/prometheus/tsdb/testutil"
+)
+
+var s *memSeries
+
+func BenchmarkHeadStripeSeriesCreate(b *testing.B) {
+	// Put a series, select it. GC it and then access it.
+	h, err := NewHead(nil, nil, nil, 1000)
+	testutil.Ok(b, err)
+	defer h.Close()
+
+	for i := 0; i < b.N; i++ {
+		s, _ = h.getOrCreate(uint64(i), labels.FromStrings("a", strconv.Itoa(i)))
+	}
+}
+
+func BenchmarkHeadStripeSeriesCreateParallel(b *testing.B) {
+	// Put a series, select it. GC it and then access it.
+	h, err := NewHead(nil, nil, nil, 1000)
+	testutil.Ok(b, err)
+	defer h.Close()
+
+	var count int64
+
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			i := atomic.AddInt64(&count, 1)
+			s, _ = h.getOrCreate(uint64(i), labels.FromStrings("a", strconv.Itoa(int(i))))
```

Comment:
this one has races when run in parallel.

btw why it is here since the PR changes seem unrelated to this benchmark.